flink-connector-jdbc/src/main/java/org/apache/flink/connector/jdbc/table/ParameterizedPredicate.java

```diff
@@ -52,8 +69,26 @@ public void setPredicate(String predicate) {
     }
 
     public ParameterizedPredicate combine(String operator, ParameterizedPredicate that) {
+        int paramIndex = String.format("(%s %s ", this.predicate, operator).length();
+        if (!that.indexesOfPredicatePlaceHolders.isEmpty()) {
+            paramIndex = paramIndex + that.indexesOfPredicatePlaceHolders.get(0);
```

Comment:
looks like a strange thin assuming that there is no more than one element inside the list

as a result more complex queries are failing , an example of query which fails
```sql
SELECT * FROM a JOIN d FOR SYSTEM_TIME AS OF a.proctime 
   ON ((d.age < 50 AND d.type = 1 AND d.age > 0) 
   OR (d.age < 70 AND d.type = 6 AND d.age > 10)) AND a.ip = d.ip
```

it fails with 
```
Caused by: java.lang.IndexOutOfBoundsException: Index: 3, Size: 3
	at java.util.ArrayList.rangeCheck(ArrayList.java:659)
	at java.util.ArrayList.get(ArrayList.java:435)
	at org.apache.flink.connector.jdbc.table.JdbcDynamicTableSource.getResolvedConditions(JdbcDynamicTableSource.java:157)
	at org.apache.flink.connector.jdbc.table.JdbcDynamicTableSource.getLookupRuntimeProvider(JdbcDynamicTableSource.java:120)
	at org.apache.flink.table.planner.plan.utils.LookupJoinUtil.createLookupRuntimeProvider(LookupJoinUtil.java:626)
	at org.apache.flink.table.planner.plan.utils.LookupJoinUtil.isAsyncLookup(LookupJoinUtil.java:413)
	at org.apache.flink.table.planner.plan.nodes.physical.common.CommonPhysicalLookupJoin.isAsyncEnabled$lzycompute(CommonPhysicalLookupJoin.scala:120)
	at org.apache.flink.table.planner.plan.nodes.physical.common.CommonPhysicalLookupJoin.isAsyncEnabled(CommonPhysicalLookupJoin.scala:116)
	at org.apache.flink.table.planner.plan.nodes.physical.common.CommonPhysicalLookupJoin.asyncOptions$lzycompute(CommonPhysicalLookupJoin.scala:129)
	at org.apache.flink.table.planner.plan.nodes.physical.common.CommonPhysicalLookupJoin.asyncOptions(CommonPhysicalLookupJoin.scala:129)
```

Reply:
@snuyanzin thanks I will look to recreate. Assuming you are ok with the general approach - I am thinking that the resolution will be to handle multiple existing indices. I will investigate.

Reply:
Please have a look at https://github.com/davidradl/flink-connector-jdbc/pull/1 where it is fixed